pkg/cmd/proxy/probe/exec.go

```diff
@@ -0,0 +1,324 @@
+	proxyv1alpha1 "open-cluster-management.io/cluster-proxy/pkg/apis/proxy/v1alpha1"
+	"open-cluster-management.io/cluster-proxy/pkg/common"
+	"open-cluster-management.io/cluster-proxy/pkg/generated/clientset/versioned"
+	"open-cluster-management.io/cluster-proxy/pkg/util"
+	konnectivity "sigs.k8s.io/apiserver-network-proxy/konnectivity-client/pkg/client"
+	proxyutil "sigs.k8s.io/apiserver-network-proxy/pkg/util"
+)
+
+func (o *Options) complete(cmd *cobra.Command, args []string) error {
+	if len(o.proxyClientCACertPath) > 0 && len(o.proxyClientCertPath) > 0 && len(o.proxyClientKeyPath) > 0 {
+		o.isProxyServerAddressProvided = true
+	}
+	if o.proxyServerHost != addrLocalhost {
+		o.isProxyServerAddressProvided = true
+	}
+	return nil
+}
+
+func (o *Options) validate() error {
+	if !o.inClusterProxyCertLookup {
+		if len(o.proxyClientCACertPath) == 0 {
+			return errors.New("--proxy-ca-cert must be set when in-cluster lookup is disabled")
+		}
+		if len(o.proxyClientCertPath) == 0 {
+			return errors.New("--proxy-cert must be set when in-cluster lookup is disabled")
+		}
+		if len(o.proxyClientKeyPath) == 0 {
+			return errors.New("--proxy-key must be set when in-cluster lookup is disabled")
+		}
+	}
+	return nil
+}
+
+func (o *Options) run(streams genericclioptions.IOStreams) error {
+
+	hubRestConfig, err := o.ClusteradmFlags.KubectlFactory.ToRESTConfig()
+	if err != nil {
+		return errors.Wrapf(err, "failed loading hub cluster's client config")
+	}
+	addonClient, err := addonv1alpha1client.NewForConfig(hubRestConfig)
+	if err != nil {
+		return errors.Wrapf(err, "failed newing addon api client")
+	}
+
+	clusterAddon, err := addonClient.AddonV1alpha1().ClusterManagementAddOns().Get(
+		context.TODO(),
+		"cluster-proxy",
+		metav1.GetOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			if _, err := fmt.Fprintf(
+				streams.Out,
+				"Cluster-Proxy addon is not installed.\n"); err != nil {
+				return err
+			}
+			if _, err := fmt.Fprintf(
+				streams.Out,
+				"Consider following the guide: https://open-cluster-management.io/getting-started/integration/cluster-proxy/\n"); err != nil {
+				return err
+			}
+			return nil
+		}
+		return errors.Wrapf(err, "failed checking cluster management addon for cluster-proxy")
+	}
+
+	proxyClient, err := versioned.NewForConfig(hubRestConfig)
+	if err != nil {
+		return errors.Wrapf(err, "failed newing proxy api client")
+	}
+	proxyConfig, err := proxyClient.ProxyV1alpha1().ManagedProxyConfigurations().
+		Get(context.TODO(), clusterAddon.Spec.AddOnConfiguration.CRName, metav1.GetOptions{})
+	if err != nil {
+		return errors.Wrapf(err, "failed getting managedproxyconfiguration for cluster-proxy")
+	}
+
+	clusterClient, err := clusterv1.NewForConfig(hubRestConfig)
+	if err != nil {
+		return errors.Wrapf(err, "failed newing cluster client")
+	}
+	managedClusterList, err := clusterClient.ManagedClusters().List(
```

Comment:
maybe list related managedclusteraddon here? Since a user may not start proxy in some clusters.

Reply:
ideally yes, but for now the automatic cluster-proxy addon installation cannot be opt-out due to framework's limitation. so every managed cluster will always install an addon constantly, what do you think if we adding the filtering after that is supported in the framework?